src/search_cuda.h

```diff
@@ -48,8 +48,10 @@ struct GreedySearch_Cuda : Search_Cuda {
   RoamingArray<int32_t> GetNextIndices() override { return gpu_span<int32_t>{}; }
 
   void SelectTop() override;
-  void SampleTopK(int k, float t) override { assert(false); }
+  void SampleTopK(int k, float t) override;
```

Comment:
why implement SampleTopK in GreedySearch?

Reply:
I think we may refactor the search into two different class: Search and BeamSearch. Search supports function: SelectTop, SelectTopK, SelectTopP, SelectTopKTopP. @RyanUnderhill